stdlib/public/core/Identifiable.swift

```diff
@@ -10,11 +10,33 @@
 //
 //===----------------------------------------------------------------------===//
 
-/// A class of types whose instances hold the value of an entity with stable identity.
+/// A class of types whose instances hold the value of an entity with stable
+/// identity.
+///
+/// `Identifiable` leaves the duration and scope of the identity unspecified.
+/// For example, identities could be:
+///
+/// - Guaranteed always unique (e.g. UUIDs).
+/// - Persistently unique per environment (e.g. database record keys).
+/// - Unique for the lifetime of a process (e.g. global incrementing integers).
+/// - Unique for the lifetime of an object (e.g. object identifiers).
+/// - Unique within the current collection (e.g. collection index).
+///
+/// It is up to both the conformer and the receiver of the protocol to document
+/// the nature of the identity.
+///
+/// Conforming to the Identifiable Protocol
+/// =======================================
+///
+/// `Identifiable` provides a default implementation for an object (its
```

Comment:
Is "object" the right terminology here? `ObjectIdentifier` and `AnyObject` use it, but we tend to clarify it in the docs:

```
/// `Identifiable` provides a default implementation for class types
/// (using `ObjectIdentifier`), ...
```